src/Handler/ContainerHandler.php

```diff
@@ -188,9 +188,11 @@ private static function addSubscribedServicesArray(Expr\Array_ $array, Codebase
                 $key = $arrayItem->key;
                 $serviceId = $key instanceof String_ ? $key->value : $className;
 
-                $service = new Service($serviceId, $className);
-                $service->setIsPublic(true);
-                self::$containerMeta->add($service);
+                if (null === self::$containerMeta->get($className)) {
```

Comment:
@bendavies thank you for the contribution. I think here, this condition can be a temporary solution. ContainerMeta and Service classes should be improved to make them work with aliases better. Here in this case, http_kernel -> HttpKernelInterface -> should resolve to HttpKernel as service.

Reply:
unsure what you mean, but `http_kernel` does resolve to `HttpKernel` in this case. see the acceptance yest

Reply:
@bendavies 

> unsure what you mean, but http_kernel does resolve to HttpKernel in this case

yes, it does because it is defined in xml file as `http_kernel`. If the name was different it wouldn't.

Lets take a little different example for better explanation:

```php
public static function getSubscribedServices(): array
{
    return [
        'http_kernel_with_a_different_name' => HttpKernelInterface::class,
    ];
}
```

In this case, this is not going to resolved with this new changes. Because `if` check is done by class name and new alias for the service is not going to be added to the container but it should. Thus it will result in service not found error.

The correct solution IMHO is to change service class to allow add more than one aliases (with visibility, public false or true) and add such cases with aliases. And on ContainerMeta class adding new service should check if it resolves to the actual class if interface is provided.